src/main/java/uk/gov/hmcts/reform/sandl.snlevents/service/VersionCheckService.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.hmcts.reform.sandl.snlevents.service;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.VersionedEntity;
+
+import javax.persistence.OptimisticLockException;
+
+@Service
+public class VersionCheckService {
```

Comment:
consider the entitymanger.merge instead http://blog.xebia.com/jpa-implementation-patterns-saving-detached-entities/